src/main/java/calculator/controller/Calculator.java

```diff
@@ -0,0 +1,47 @@
+package calculator.controller;
+
+import calculator.domain.calculator.CalculatorNumberFactory;
+import calculator.domain.calculator.CalculatorNumbers;
+import calculator.domain.delimeter.DelimiterFactory;
+import calculator.domain.delimeter.Delimiters;
+import calculator.dto.ExtractionResult;
+import calculator.service.DelimiterExtractor;
+import calculator.service.StringSplitter;
+import calculator.view.InputView;
+import calculator.view.OutputView;
+import java.util.List;
+
+public class Calculator {
+    private final InputView inputView;
+    private final OutputView outputView;
+    private final DelimiterExtractor delimiterExtractor;
+    private final StringSplitter stringSplitter;
+
+    public Calculator() {
+        this.inputView = new InputView();
+        this.outputView = new OutputView();
+        this.delimiterExtractor = new DelimiterExtractor();
+        this.stringSplitter = new StringSplitter();
+    }
```

Comment:
코드리뷰를 하면서 이런 패턴을 자주 보는것 같아요.
저같은 경우에는 
private InputView inputView = new InputView();
와 같은 형식으로 코드를 작성했는데,
생성자를 사용하면 어떤 이점이 있는건가요???

Reply:
@CodingMasterLSW 
우선 지금같은 경우에는 생성자 내부에서 new로 객체를 생성하는 방식은 필드에서 바로 생성하는 것과 큰 차이가 없습니다.

```suggestion
    private final InputView inputView;
    private final OutputView outputView;
    private final DelimiterExtractor delimiterExtractor;
    private final StringSplitter stringSplitter;

    public Calculator(
        InputView inputView, 
        OutputView outputView,
        DelimiterExtractor delimiterExtractor,
        StringSplitter stringSplitter
    ) {
        this.inputView = inputView;
        this.outputView = outputView;
        this.delimiterExtractor = delimiterExtractor;
        this.stringSplitter = stringSplitter;
    }
```
만약 제가 이런 방식으로 외부에서 객체를 생성해 주입받았다면 결합도가 훨씬 낮아지고, 더 유연한 구조를 만들 수 있었을 것 같습니다. 객체의 변경이나 확장 시, 또는 테스트 시 모의 객체(mock object)를 사용하기에 더 용이합니다.